shared/fs/row.js

```diff
@@ -3,44 +3,45 @@ import * as Types from '../constants/types/fs'
 import {connect, type TypedState, type Dispatch} from '../util/container'
 import {isMobile} from '../constants/platform'
 import {navigateAppend} from '../actions/route-tree'
+import type {IconType} from '../common-adapters/icon'
 
 type OwnProps = {
   path: Types.Path,
 }
 
-type StateProps = {
-  type: Types.PathType,
+type DispatchProps = {
+  _onOpen: (type: Types.PathType, path: Types.Path) => void,
 }
 
-type DispatchProps = {
-  _onViewFolder: Types.Path => void,
-  _onViewFile: Types.Path => void,
+const mapStateToProps = (state: TypedState, {path}: OwnProps) => {
+  const pathItem = state.fs.pathItems.get(path)
+  return {path, type: pathItem ? pathItem.type : 'unknown'}
 }
 
-const mapStateToProps = ({fs}: TypedState, {path}: OwnProps) => ({
-  type: Types.stringToPathType(fs.pathItems.getIn([path, 'type'], '')),
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _onOpen: (type: Types.PathType, path: Types.Path) => {
```

Comment:
if we're going to switch we might as well do this here instead of mergeProps, keep that more of a simpler aggregator of props